### ISSUE TITLE
Extend  `--with-attribute-custom` to support more types

### ISSUE THREAD

**Issue by jschwe** [rust-lang/rust-bindgen#2978]
I would like to use custom attributes to feature guard bindings based on information in the doc comment.

E.g. if the doc comment contains `@since 10`, i would like to generate 
```
#[cfg(feature = "api-10")]
<the item>
```

Currently `add_attributes` seems to only support structs, enums and unions. I would like to also have support for functions and ideally also enum variants.
Would a PR for this be accepted?

CC @mkroening Any technical reason your PR did not implement support for functions, or did you just not need it for your usecase?

**Comment by mkroening** [comment]
> @mkroening Any technical reason your PR did not implement support for functions, or did you just not need it for your usecase?

Nope, no technical reason, I just built off the variants of `--with-derive-custom`, which do not support functions.

**Comment by pvdrz** [comment]
I think this is doable, it only needs to get proper support
